### PR TITLE
Fix: push results to GCS

### PR DESCRIPTION
**Pull request by OlegLoewen**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

File: test/k8s-e2e/e2e-test-execute [hunk 1]
```diff
@@ -355,10 +358,13 @@ function run_e2e_tests_kubetest() {
 
 	done 3<<<"$(cat "${generated_run_desc_path}")"
 
-	push_test_summary_to_elastic_search
+	save_test_results_summary_as_file
 
 	if [[ "$publish_test_results" -eq "$TRUE" ]]; then
-		pushResutlsToGCS
+		if ! pushResutlsToGCS; then
```
Comment (dguendisch) [inline test/k8s-e2e/e2e-test-execute]: typo
```suggestion
		if ! pushResultsToGCS; then
```

File: test/k8s-e2e/e2e-test-execute [hunk 2]
```diff
@@ -397,7 +403,6 @@ function gather_test_summary_data_for_elastic_search() {
 
 function pushResutlsToGCS() {
```
Comment (dguendisch) [inline test/k8s-e2e/e2e-test-execute]: typo
```suggestion
function pushResultsToGCS() {
```

File: test/k8s-e2e/convtojson.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def convert_junit_01_xml_to_json(
                 res['name'] = res['name'].strip()
                 matchObj = re.match(r'\[(.*?)\].*', res['name'], re.M | re.I)
                 res['sig'] = matchObj.group(1).strip()
-                res['duration'] = float(res['time'])
+                res['duration'] = int(float(res['time']))
```
Comment (schrodit) [inline test/k8s-e2e/convtojson.py]: Why do you need a number not just leaving the string?

File: test/k8s-e2e/e2e-test-execute [hunk 4]
```diff
@@ -316,7 +316,10 @@ function run_e2e_tests_kubetest() {
 		return $success
 	fi
 
-	isKubeAPIReachable
+	if ! isKubeAPIReachable; then
+		success="$FALSE"
+		return $success
```
Comment (schrodit) [inline test/k8s-e2e/e2e-test-execute]: return immediately without assigning the success var?

File: test/k8s-e2e/e2e-test-execute [hunk 5]
```diff
@@ -430,15 +428,15 @@ function pushResutlsToGCS() {
...
 	if [ "$CLOUDPROVIDER" = "gcp" ]; then
 		CLOUDPROVIDER_PUBLIC="gce"
 	elif [[ "$CLOUDPROVIDER" = os-* ]]; then
 		CLOUDPROVIDER_PUBLIC="os"
 	else
 		CLOUDPROVIDER_PUBLIC=${CLOUDPROVIDER}
 	fi
-		upload_res="$(python2 ./test-infra/testgrid/conformance/upload_e2e.py --junit="$log_dir/junit_01.xml" --log="$e2e_log" --bucket=gs://k8s-conformance-gardener/ci-gardener-e2e-conformance-"${CLOUDPROVIDER_PUBLIC}"-v"${K8S_RELEASE%*\.[0-9]*}" --metadata='{"commit-id":"'$commit_id'","shoot-k8s-release":"'$K8S_RELEASE'"}' --key-file=$google_bucket_secrets_file)"
+
+	upload_res="$(python2 ./test-infra/testgrid/conformance/upload_e2e.py --junit="$log_dir/junit_01.xml" --log="$e2e_log" --bucket=gs://k8s-conformance-gardener/ci-gardener-e2e-conformance-"${CLOUDPROVIDER_PUBLIC}"-v"${K8S_VERSION%*\.[0-9]*}" --metadata='{"commit-id":"'$GARDENER_COMMIT_ID'","shoot-k8s-release":"'$K8S_VERSION'"}' --key-file=$google_bucket_secrets_file)"
```
Comment (schrodit) [inline test/k8s-e2e/e2e-test-execute]: Do you need a commit ID or is a version ok?

File: test/k8s-e2e/e2e-test-execute (hunk 4 above)
Comment (schrodit) [reply test/k8s-e2e/e2e-test-execute]: also success ist already set by the `isKubeAPIReachable` function (https://github.com/gardener/test-infra/blob/master/test/k8s-e2e/e2e-test-execute#L248)

File: test/k8s-e2e/convtojson.py (hunk 3 above)
Comment (OlegLoewen) [reply test/k8s-e2e/convtojson.py]: removing number after the decimal point makes the time values more readable.

File: test/k8s-e2e/e2e-test-execute (hunk 5 above)
Comment (OlegLoewen) [reply test/k8s-e2e/e2e-test-execute]: doesn't matter which. This is just additional optional information to comprehend the results.

**Comment by OlegLoewen** [comment]
Requested changes applied. Pleas rereview

**Pull request merged**